run.go

```diff
@@ -326,38 +329,96 @@ func Log(c chan *SimulationStatus) DomainManipulator {
 	}
 }
 
-func (d *InMAP) checkOutputNames(n ...string) error {
+func removeDuplicates(s []string) []string {
+	result := []string{}
+	seen := map[string]string{}
+	for _, val := range s {
+		if _, ok := seen[val]; !ok {
+			result = append(result, val)
+			seen[val] = val
+		}
+	}
+	return result
+}
+
+// Checks if outputVariables are defined in terms of other outputVariables.
+// Creates a slice of unique model variables that are required to calculate
+// the requested outputVariables.
+func checkForDerivitives(m map[string]string) (map[string]string, []string) {
+	getVariables := []string{}
+	for key, val := range m {
+		expression, _ := govaluate.NewEvaluableExpression(val)
+		uniqueVars := removeDuplicates(expression.Vars())
```

Comment:
This line will need to be updated because the "Vars()" method is only available in the version of the GoValuate package that David Paolella edited. An alternate approach needs to be found that utilizes the official GoValuate package